### PR TITLE
Clean up HTTP service configuration with new Armeria API

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

The Central Dogma HTTP service configurations used old Armeria API and had many duplicate code.

Modifications:

- Use the context-path API to register API services.
- Use a route decorator to not bind the decorator to each service
- Use `ServerErrorHandler` to apply it globally
- Use `DependencyInjector` to inject request/response converters with annotations.

Result:

Duplicate code has been removed and streamlined

### CODE REVIEW

File: server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java [hunk 1]
```diff
@@ -620,6 +621,9 @@ private Server startServer(ProjectManager pm, CommandExecutor executor,
                          meterRegistry);
 
         configureMetrics(sb, meterRegistry);
+        // Add the CORS service as the last decorator(executed first) so that the CORS service is applied
+        // before AuthService.
+        configCors(sb, config().corsConfig());
```
Comment (minwoox) [inline server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java:626]: 👍

File: server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java [hunk 2]
```diff
@@ -773,37 +777,45 @@ protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) {
...
-                                new MirroringServiceV1(projectApiManager, executor), decorator,
-                                v1RequestConverter, jacksonRequestConverterFunction, v1RequestConverter);
-            sb.annotatedService(API_V1_PATH_PREFIX,
-                                new CredentialServiceV1(projectApiManager, executor), decorator,
-                                v1RequestConverter, jacksonRequestConverterFunction, v1RequestConverter);
+            apiV1ServiceBuilder.annotatedService(new MirroringServiceV1(projectApiManager, executor))
+                               .annotatedService(new CredentialServiceV1(projectApiManager, executor));
         }
 
+        // TODO(ikhoon): Use context-path API once https://github.com/line/armeria/pull/5797 is fixed.
```
Comment (minwoox) [inline server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java:818]: 👍

**Review by minwoox — APPROVED** [review]
Looks much cleaner. 👍 👍 👍

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

File: server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java (hunk 2 above)
Comment (minwoox) [reply server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java]: It's done. 😉

**Comment by minwoox** [comment]
👍 👍 👍

**Pull request merged**